### PR TITLE
Add support for `PendingUpdate` and `ProrationBehavior` on `Subscription` APIs.

### PR DESCRIPTION
**Pull request by cjavilla-stripe**
I don't feel super confident in the type for `SubscriptionItems` on PendingUpdate here and am just trying to wrap up testing of deserialization locally.

r? @remi-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/Stripe.net/Entities/Subscriptions/Subscription.cs [hunk 1]
```diff
@@ -45,9 +45,24 @@ public class Subscription : StripeEntity<Subscription>, IHasId, IHasMetadata, IH
         [JsonConverter(typeof(DateTimeConverter))]
         public DateTime? CancelAt { get; set; }
 
+        /// <summary>
+        /// If the subscription has been canceled with the <c>AtPeriodEnd</c> flag
```
Comment (remi-stripe) [inline src/Stripe.net/Entities/Subscriptions/Subscription.cs]: Usually this is a CRef

File: src/Stripe.net/Entities/Subscriptions/Subscription.cs [hunk 2]
```diff
@@ -45,9 +45,24 @@ public class Subscription : StripeEntity<Subscription>, IHasId, IHasMetadata, IH
...
+        /// set to true, <c>CancelAtPeriodEnd</c> on the subscription will be true.
```
Comment (remi-stripe) [inline src/Stripe.net/Entities/Subscriptions/Subscription.cs]: same here

I won't do all of them, we might just be okay with this for now. Let's talk to @ob-stripe on Monday

File: src/Stripe.net/Entities/Subscriptions/Subscription.cs [hunk 3]
```diff
@@ -281,6 +296,13 @@ public SetupIntent PendingSetupIntent
         internal ExpandableField<SetupIntent> InternalPendingSetupIntent { get; set; }
         #endregion
 
+        /// <summary>
+        /// If specified, deferred upgrade changes that will be applied to the
+        /// subscription once the latest_invoice has been paid.
```
Comment (remi-stripe) [inline src/Stripe.net/Entities/Subscriptions/Subscription.cs]: missing CRef or <c>

File: src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        [JsonProperty("expires_at")]
+        [JsonConverter(typeof(DateTimeConverter))]
+        public DateTime? ExpiresAt { get; set; }
+
+        /// <summary>
+        /// List of subscription items, each with an attached plan, that will
+        /// be set if the update is applied.
+        /// </summary>
+        [JsonProperty("subscription_items")]
+        public StripeList<SubscriptionItem> SubscriptionItems { get; set; }
```
Comment (remi-stripe) [inline src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs]: Shouldn't this be a `List` and not a `StripeList`?

Comment (cjavilla-stripe) [reply src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs]: Yeah I think you're right. This was one of the things I wanted to test deserializing with stripe-mock.

**Comment by cjavilla-stripe** [comment]
@ob-stripe when you get a chance. I'm generally curious how you think we should handle the `see cref` references in the doc comments once we have auto-gen. Curious if we should always use `<see cref="TitleCasedThing" />`? or if we need to some how know when to use `<see cref` and `<c>` depending on context

**Comment by remi-stripe** [comment]
also it's missing the constants for the events right?

**Comment by cjavilla-stripe** [comment]
Good catch on events, @remi. I noticed it here: https://github.com/stripe/stripe-java/pull/925/files but not in the latest stripe-java PR: https://github.com/stripe/stripe-java/pull/933/files should that be in the latest?

I tried testing out the `<see cref` feature with visual studios and wasn't seeing how this actually works.

For instance, here's Plan.BillingScheme's docs, which don't seem to link to the properties?

image

**Comment by cjavilla-stripe** [comment]
updated and pretty sure I got the comment references this time.

r? @remi-stripe

File: src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        [JsonConverter(typeof(DateTimeConverter))]
+        public DateTime? BillingCycleAnchor { get; set; }
+
+        /// <summary>
+        /// The point after which the changes reflected by this update will be
+        /// discarded and no longer applied.
+        /// </summary>
+        [JsonProperty("expires_at")]
+        [JsonConverter(typeof(DateTimeConverter))]
+        public DateTime? ExpiresAt { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs]: According to the spec, this field isn't nullable.

File: src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        public DateTime? TrialEnd { get; set; }
+
+        /// <summary>
+        /// Indicates if a <see cref="Plan.TrialPeriodDays"/> should
+        /// be applied to the subscription. Setting <see cref="TrialEnd"/> per
+        /// subscription is preferred, and this defaults to false. Setting this
+        /// flag to true together with <see cref="TrialEnd"/> is not allowed.
+        /// </summary>
+        [JsonProperty("trial_from_plan")]
+        public bool TrialFromPlan { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Subscriptions/SubscriptionPendingUpdate.cs]: And this field is nullable.

**Comment by cjavilla-stripe** [comment]
Flipped the nullability of those two columns and bumped stripe-mock. Thanks for looking @ob-stripe 

r? @ob-stripe

**Pull request merged**